src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -316,7 +426,13 @@ private Path download(File installDirectory, HttpUrl downloadUrl, String current
         headers.put(DOWNLOAD_VERSION_HEADER, currentVersion);
         final Request request = new Request(downloadUrl, HttpMethod.GET, 
                 null, queryParams, headers, null);
-        
+        final ProxyInfo proxyInfo;
```

Comment:
Here, we check if the BD host is on the ignore-host-for-proxy list or we have not found any proxy settings configured. Then, we opt for No Proxy. Otherwise, we prepare the proxy information from the identified proxy settings.